pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -54,14 +54,15 @@ declare -gi depCacheInitialised=0
 declare -gi doneRecursiveSearch=0
 declare -g foundDependency
 
-getDepsFromSo() {
-    ldd "$1" 2> /dev/null | sed -n -e 's/[^=]*=> *\(.\+\) \+([^)]*)$/\1/p'
+getDepsFromElfBinary() {
+    $READELF -d "$1" 2> /dev/null | \
+        sed -n -e 's/^[\t ]*0x[0-9a-f]\{1,\}[\t ]\{1,\}(NEEDED)[\t ]\{1,\}Shared library: \[\([^]]\{1,\}\)\]/\1/p'
```

Comment:
This regex needs a comment.

Reply:
I switched this function to just use `patchelf --print-needed` instead which gives exactly the same output.